Rakefile

```diff
@@ -1,81 +1,41 @@
-require "json"
 require "pathname"
 
-CONFIG_FILE  = Pathname.new(__dir__).join(ENV["CNFFILE"] || "config.json")
-BUILD_ROOT   = Pathname.new(ENV["OBJDIR"] || "build")
+HERE  = Pathname.new(".")
 INSTALL_ROOT = Pathname.new(ENV["PREFIX"] || "/opt/ood")
+VENDOR_BUNDLE_DIR = Pathname.new(ENV["VENDOR_BUNDLE_DIR"] || "../../vendor/bundle")
 
-def all_components
-  JSON.parse(CONFIG_FILE.read).map { |c| Component.new(c) }
-end
-
-class Component
-  attr_reader :name
-  attr_reader :url
-  attr_reader :tag
-
-  def initialize(opts = {})
-    @name = opts.fetch("name") { raise ArgumentError, "No name specified. Missing argument: name" }.to_s
-    @url  = opts.fetch("url")  { raise ArgumentError, "No url specified. Missing argument: url" }.to_s
-    @tag  = opts.fetch("tag")  { raise ArgumentError, "No tag specified. Missing argument: tag" }.to_s
-    @app  = opts.fetch("app", nil)
-  end
-
-  def app?
-    !!@app
-  end
-
-  def build_root
-    app? ? BUILD_ROOT.join("apps", name) : BUILD_ROOT.join(name)
-  end
-
-  def download_url
-    "#{url}/archive/#{tag}.tar.gz"
-  end
-
-  def file_url
-    "apps/#{name}/."
-  end
+def apps
+  %w(activejobs bc_desktop dashboard file-editor files myjobs shell).map {|name| HERE.join("apps", name) }
 end
 
 task :default => :build
 
-all_components.each do |c|
-  file c.build_root => CONFIG_FILE do
-    rm_rf c.build_root if c.build_root.directory?
-    mkdir_p c.build_root unless c.build_root.directory?
-    cp_r c.file_url, c.build_root
-    setup_path = c.build_root.join("bin", "setup")
-    if setup_path.exist? && setup_path.executable?
-      sh "PASSENGER_APP_ENV=production PASSENGER_BASE_URI=/pun/sys/#{c.name} #{setup_path}"
-    end
-    c.build_root.join("VERSION").write(c.tag) if c.app?
-  end
-end
-
-def proxy_components
-  %w(ood-portal-generator mod_ood_proxy ood_auth_map nginx_stage).map {|name| BUILD_ROOT.join(name) }
-end
-
-proxy_components.each do |build_root|
-  file build_root => 'nginx_stage/lib/nginx_stage/version.rb' do
-    rm_rf build_root if build_root.directory?
-    cp_r build_root.basename, build_root
+apps.each do |app|
+  setup_path = app.join("bin", "setup")
+  if setup_path.exist? && setup_path.executable?
+    sh "PASSENGER_APP_ENV=production VENDOR_BUNDLE_DIR=#{VENDOR_BUNDLE_DIR} #{setup_path}"
   end
 end
```

Comment:
This block iterates over each app and builds it. Instead, this block should iterate over each app and create a task that will build the app when executed.

For example, running `rake -T` displays all the tasks available, but should not execute any individual task. If you run `rake -T` here, all the apps get built.

Reply:
Actually doing something like this would be preferable:

```
namespace :build do
  desc "build app 1"
  task "app1" do
    puts "build app 1"
  end

  desc "build app 2"
  task "app2" do
    puts "build app 2"
  end

  desc "build all apps"
  task :all => [:app1, :app2]
end

desc "build all apps"
task :build => 'build:all'
```

except of course you have a task for each app, with the name of the app (without the full path)

That gives you something like this:

```ruby
➜  r rake -T
rake build:app1  # build app 1
rake build:app2  # build app 2
rake builds      # build all apps
```